src/dbos-executor.ts

```diff
@@ -1921,19 +1886,27 @@ export class DBOSExecutor implements DBOSExecutorContext {
     }
   }
 
-  async getWorkflowStatus(workflowID: string, callerID?: string, callerFN?: number): Promise<WorkflowStatus | null> {
+  async getWorkflowStatus(
+    workflowID: string,
+    callerID?: string,
+    callerFN?: number,
+    getRequest: boolean = false,
+  ): Promise<WorkflowStatus | null> {
+    // use sysdb getWorkflowStatus directly in order to support caller ID/FN params
     const status = await this.systemDatabase.getWorkflowStatus(workflowID, callerID, callerFN);
-    return status ? DBOSExecutor.toWorkflowStatus(status, true) : null;
+    return status ? toWorkflowStatus(status, getRequest) : null;
   }
 
-  async listWorkflows(input: GetWorkflowsInput): Promise<WorkflowStatus[]> {
-    const wfs = await this.systemDatabase.listWorkflows(input);
-    return wfs.map((wf) => DBOSExecutor.toWorkflowStatus(wf, true));
+  async listWorkflows(input: GetWorkflowsInput, getRequest: boolean = false): Promise<WorkflowStatus[]> {
+    return listWorkflows(this.systemDatabase, input, getRequest);
   }
 
-  async listQueuedWorkflows(input: GetQueuedWorkflowsInput): Promise<WorkflowStatus[]> {
-    const wfs = await this.systemDatabase.listQueuedWorkflows(input);
-    return wfs.map((wf) => DBOSExecutor.toWorkflowStatus(wf, true));
+  async listQueuedWorkflows(input: GetQueuedWorkflowsInput, getRequest: boolean = false): Promise<WorkflowStatus[]> {
+    return listQueuedWorkflows(this.systemDatabase, input, getRequest);
+  }
```

Comment:
same comment about `request`

Reply:
same response - good suggestion not to add it to places where it wasn't already there